superset/views/base.py

```diff
@@ -380,7 +380,9 @@ def menu_data(user: User) -> dict[str, Any]:
 
 
 @cache_manager.cache.memoize(timeout=60)
-def cached_common_bootstrap_data(user: User, locale: str) -> dict[str, Any]:
+def cached_common_bootstrap_data(  # pylint: disable=unused-argument
+    user: User, user_id: int | None, locale: str
```

Comment:
Why do we need both `User` and `user_id`? Surely the former contains the ID which should then be used for the cache key given it's globally unique.

Reply:
While yes, `User` "contains" the ID, it's `__repr__` method that flask-caching uses to generate the cache key does not (see PR description).

The 2 alternatives to this fix as-is would be to:
1. change the `__repr__` upstream on FAB to include the id. (cc @dpgaspar if this is a welcome change)
2. Subclass FAB's user class and add our own __repr__ then update to use that user class everywhere, which seems like a pretty large change for this small fix

Reply:
@jfrag1 sorry I missed that detail when initially glancing over the PR description. Is a third alternative to just pass in `user_id` instead of both `user` and `user_id`?